### PR TITLE
Change SO pp multiplier to be based on amount of spinners in the map

### PR DESCRIPTION
**Pull request by stanriders**
https://github.com/ppy/osu/pull/6892

### CODE REVIEW

File: include/pp/performance/Beatmap.h [hunk 1]
```diff
@@ -46,11 +46,13 @@ class Beatmap
 	ERankedStatus RankedStatus() const { return _rankedStatus; }
 	EScoreVersion ScoreVersion() const { return _scoreVersion; }
 	s32 NumHitCircles() const { return _numHitCircles; }
+	s32 NumSpinners() const { return _numHitCircles; }
```
Comment (pavlukivan) [inline include/pp/performance/Beatmap.h]: should return _numSpinners

Comment (stanriders) [reply include/pp/performance/Beatmap.h]: fixed

**Comment by smoogipoo** [comment]
@peppy requesting a dump of a significant number of scores with SO mod enabled (maybe like top 10000?)

**Comment by stanriders** [comment]
https://newpp.stanr.info/user/so (5.5 mb!) I've calculated all SO scores from top10k database (22k total), but now I think I should've compared it to nomod plays instead of live. Don't mind the website, values are from the clean diffcalc, I'm just using the website for all pp experiments nowadays

File: src/performance/osu/OsuScore.cpp [hunk 2]
```diff
@@ -70,8 +70,9 @@ void OsuScore::computeTotalValue()
 	if ((_mods & EMods::NoFail) > 0)
 		multiplier *= 0.90f;
 
+	int numTotalHits = TotalHits();
 	if ((_mods & EMods::SpunOut) > 0)
-		multiplier *= 0.95f;
+		multiplier *= 1.0f - std::pow(beatmap.NumSpinners() / static_cast<f32>(numTotalHits), 0.85f);
```
Comment (smoogipoo) [inline src/performance/osu/OsuScore.cpp:75]: Is this really doing much? It's basically almost a linear function:
image

With the delta being ~0.06 at most. Do you have anything against making this just the following for simplicity?

```suggestion
		multiplier *= 1.0f - beatmap.NumSpinners() / static_cast<f32>(numTotalHits);
```

Comment (stanriders) [reply src/performance/osu/OsuScore.cpp:75]: I believe it was to make it closer to current 0.95 multiplier for maps with enough spinners

Comment (smoogipoo) [reply src/performance/osu/OsuScore.cpp:75]: Sure, I guess it's not a big deal...

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**